### DISCUSSION TITLE
How does it work?

### DISCUSSION THREAD

**Discussion by zimmerma379** [Mygod/VPNHotspot#457] [Q&A]
So glad I found this app!

I had the following set of commands, as I understood them, would route my phone’s hotspot traffic thru my VPN; and this would bypass my carrier’s throttling of my phone’s hotspot clients:
    # iptables -t filter -F FORWARD
    # iptables -t nat -F POSTROUTING
    # iptables -t filter -I FORWARD -j ACCEPT
    # iptables -t nat -I POSTROUTING -j MASQUERADE
    # ip rule add from 192.168.43.0/24 lookup 61
    # ip route add default dev tun0 scope link table 61
    # ip route add 192.168.43.0/24 dev wlan0 scope link table 61
    # ip route add broadcast 255.255.255.255 dev wlan0 scope link table 61
But at some point this stopped working and my hotspot clients were throttled.  I wonder how my carrier now detects this workaround?

Luckily, I found “VPN Hotspot” and am able to once again bypass my carrier’s throttling of my phone’s hotspot clients.  I’m so curious how the “VPN Hotspot” set of commands differs from the set of commands above.  I’m guessing “VPN Hotspot” takes a more complex approach, involving a “softap0” virtual device?  I browsed the open source for a bit, but didn’t have time to dive in deeper.  I don’t understand much about IP routing, so if anyone can educate me on the set of commands issued by “VPN Hotspot” I’d appreciate it.  Thanks!

**Comment by Mygod** [comment]
Roughly the same way but more stable and more reliable.
